drivers/serial/uart_stm32.c

```diff
@@ -341,6 +341,10 @@ static int uart_stm32_configure(struct device *dev,
 	}
 
 	if (cfg->baudrate != data->baud_rate) {
+		if (IS_LPUART_INSTANCE(UartInstance) &&
+			(data->baud_rate > 9600)) {
+				return -ENOTSUP;
+			}
 		uart_stm32_set_baudrate(dev, cfg->baudrate);
```

Comment:
Shouldn't the '}' be aligned with the 'if' ?
Identation apart, the logic seems good.